state-chain/pallets/cf-elections/src/electoral_systems/mocks/access.rs

```diff
@@ -114,183 +69,387 @@ macro_rules! impl_read_access {
 				previous_consensus: Option<&ES::Consensus>,
 				votes: ConsensusVotes<ES>,
 			) -> Result<Option<ES::Consensus>, CorruptStorageError> {
-				ES::check_consensus(self.identifier(), self, previous_consensus, votes)
+				println!("Calling check consensus on the electoral system struct");
+				ES::check_consensus(self, previous_consensus, votes)
 			}
 		}
 	};
 }
 
-impl_read_access!(MockReadAccess<'_, ES>);
-impl_read_access!(MockWriteAccess<'_, ES>);
+impl_read_access!(MockReadAccess<ES>);
+impl_read_access!(MockWriteAccess<ES>);
 
-impl<ES: ElectoralSystem> MockWriteAccess<'_, ES> {
-	fn with_election_mut<F: FnOnce(&mut MockElection<ES>) -> R, R>(
-		&mut self,
-		f: F,
-	) -> Result<R, CorruptStorageError> {
-		self.electoral_system
-			.elections
-			.get_mut(&self.identifier())
-			.map(f)
-			.ok_or_else(CorruptStorageError::new)
-	}
-	pub fn set_consensus_status(&mut self, consensus_status: ConsensusStatus<ES::Consensus>) {
-		self.with_election_mut(|e| e.consensus_status = consensus_status)
-			.expect("Cannot set consensus status for non-existent election");
-	}
+thread_local! {
+	pub static ELECTION_STATE: RefCell<BTreeMap<Vec<u8>, Vec<u8>>> = const { RefCell::new(BTreeMap::new()) };
+	pub static ELECTION_PROPERTIES: RefCell<BTreeMap<Vec<u8>, Vec<u8>>> = const { RefCell::new(BTreeMap::new()) };
+	pub static ELECTORAL_SETTINGS: RefCell<Vec<u8>> = const { RefCell::new(Vec::new()) };
```

Comment:
Not sure we want to use thread_locals here - what if we rely on storage access properties such as transactional access? (Don't believe we do rn. but seems worth supporting). Also things like assert_noop won't work (I think it will always succeed as long as there is an error because storage will never be touched). 

It should be simple enough to changes this using parameter_types:

```rust
parameter_types! {
	storage A: T = default_value
}
```

(And in fact we could have just used the old `MockAccess` struct and stored it instead of fragmenting everything again? Would have saved a lot of boilerplate?)